From: David Redondo <qt@david-redondo.de>
Date: Wed, 12 Oct 2022 09:01:07 +0200
Subject: [PATCH] Implement QNativeInterface::Private::QWaylandScreen

--- a/src/client/qwaylandscreen_p.h
+++ b/src/client/qwaylandscreen_p.h
@@ -34,7 +35,10 @@ public:
     QWaylandXdgOutputManagerV1(QWaylandDisplay *display, uint id, uint version);
 };
 
-class Q_WAYLANDCLIENT_EXPORT QWaylandScreen : public QPlatformScreen, QtWayland::wl_output, QtWayland::zxdg_output_v1
+class Q_WAYLANDCLIENT_EXPORT QWaylandScreen : public QPlatformScreen,
+                                              QtWayland::wl_output,
+                                              QtWayland::zxdg_output_v1,
+                                              public QNativeInterface::Private::QWaylandScreen
 {
 public:
     QWaylandScreen(QWaylandDisplay *waylandDisplay, int version, uint32_t id);
