Author: Lu YaNing <luyaning@uniontech.com>
Date:   Thur, 16 May 2024 10:45:02 +0800
Subject: Use static_cast rather than dynamic_cast
Upstream: https://codereview.qt-project.org/c/qt/qtwayland/+/547757


--- qt6-wayland-6.6.1.orig/src/client/qwaylanddisplay.cpp
+++ qt6-wayland-6.6.1/src/client/qwaylanddisplay.cpp
@@ -471,13 +471,13 @@ void QWaylandDisplay::reconnect()
 
     const auto windows = QGuiApplication::allWindows();
     for (auto window : windows) {
-        if (auto waylandWindow = dynamic_cast<QWaylandWindow *>(window->handle()))
+        if (auto waylandWindow = static_cast<QWaylandWindow *>(window->handle()))
             waylandWindow->closeChildPopups();
     }
     // Remove windows that do not need to be recreated and now closed popups
     QList<QWaylandWindow *> recreateWindows;
     for (auto window : std::as_const(windows)) {
-        auto waylandWindow = dynamic_cast<QWaylandWindow*>((window)->handle());
+        auto waylandWindow = static_cast<QWaylandWindow*>((window)->handle());
         if (waylandWindow && waylandWindow->wlSurface()) {
             waylandWindow->reset();
             recreateWindows.push_back(waylandWindow);
