Author: Tian Shilin<tianshilin@uniontech.com>
Date: Thu May 07 10:20:24 2026
Subject: Notify application code of button release after leaves
Upstream: https://codereview.qt-project.org/c/qt/qtwayland/+/581714

---

Index: qt6-wayland/src/client/qwaylandinputdevice.cpp
===================================================================
--- qt6-wayland.orig/src/client/qwaylandinputdevice.cpp
+++ qt6-wayland/src/client/qwaylandinputdevice.cpp
@@ -894,15 +894,7 @@ void QWaylandInputDevice::Pointer::inval
 
 void QWaylandInputDevice::Pointer::releaseButtons()
 {
-    if (mButtons == Qt::NoButton)
-        return;
-
-    mButtons = Qt::NoButton;
-
-    if (auto *window = focusWindow()) {
-        ReleaseEvent e(focusWindow(), mParent->mTime, mSurfacePos, mGlobalPos, mButtons, mLastButton, mParent->modifiers());
-        window->handleMouse(mParent, e);
-    }
+    setFrameEvent(new ReleaseEvent(nullptr, mParent->mTime, mSurfacePos, mGlobalPos, Qt::NoButton, Qt::NoButton, mParent->modifiers()));
 }
 
 void QWaylandInputDevice::Pointer::leavePointers()
